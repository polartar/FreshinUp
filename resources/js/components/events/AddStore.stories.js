import { storiesOf } from '@storybook/vue'

import AddStore from './AddStore'

const stores = [
  {
    uuid: 'a7936425-485a-4419-9acd-13cdccaed346',
    name: 'Burger Babes',
    type: {
      id: 1,
      name: 'Mobil'
    },
    location: {
      id: 1,
      uuid: 'c4fd0928-b7eb-43ee-871e-e63bfbd0ae7a',
      name: 'Port Gerald'
    },
    store_tags: [
      {
        uuid: '1',
        name: 'minus'
      },
      {
        uuid: '2',
        name: 'hic'
      }
    ],
    assigned: false
  },
  {
    uuid: 'c48fb5d3-37e0-4cb5-bb44-d2d1b5fd97d5',
    name: 'Holy Canoli',
    type: {
      id: 1,
      name: 'Mobil'
    },
    location: {
      id: 2,
      uuid: '00aabd70-aa77-42de-87fd-96449bbd5439',
      name: 'Carminestad'
    },
    store_tags: [
      {
        uuid: '1',
        name: 'minus'
      },
      {
        uuid: '2',
        name: 'sit'
      },
      {
        uuid: '3',
        name: 'hicsdfsdf'
      }
    ],
    assigned: true
  },
  {
    uuid: '790aba97-1eb6-4630-82d9-7bd561256c67',
    name: 'Da Lobsta',
    type: {
      id: 1,
      name: 'Mobil'
    },
    location: {
      id: 2,
      uuid: 'dfb4bd02-c298-4dd4-ab49-2c3f156f2894',
      name: 'New Charlenebury'
    },
    store_tags: [
      {
        uuid: '1',
        name: 'minus'
      },
      {
        uuid: '2',
        name: 'hic'
      },
      {
        uuid: '3',
        name: 'acsfdd'
      },
      {
        uuid: '4',
        name: 'fsdf'
      }
    ],
    assigned: false
  }
]

storiesOf('FoodFleet|event/AddStore', module)
  .add('empty', () => ({
    components: { AddStore },
    template: `
      <v-container>
        <add-store />
      </v-container>
    `
  }))
  .add('with data', () => ({
    components: { AddStore },
    data () {
      return {
        stores: stores
      }
    },
    template: `
      <v-container>
        <add-store :members="stores"/>
      </v-container>
    `
  }))
