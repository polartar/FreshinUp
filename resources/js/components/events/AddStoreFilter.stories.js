import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import MockAdapter from 'axios-mock-adapter'
import axios from 'axios/index'

// Components
import AddStoreFilter from './AddStoreFilter.vue'

const mock = new MockAdapter(axios)
mock.onGet('foodfleet/store-tags').reply(200, {
  data: [
    { uuid: 1, name: 'aperiam' },
    { uuid: 2, name: 'iure' },
    { uuid: 3, name: 'dicta' },
    { uuid: 4, name: 'voluptate' }
  ]
})

mock.onGet('foodfleet/locations').reply(200, {
  data: [
    { uuid: 1, name: 'South Abagail' },
    { uuid: 2, name: 'Lindseymouth' },
    { uuid: 3, name: 'Fredrickstad' },
    { uuid: 4, name: 'Zanderstad' }
  ]
})

storiesOf('FoodFleet|event/AddStoreFilter', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('default', () => ({
    components: { AddStoreFilter },
    data () {
      return {
        types: [
          { uuid: 1, name: 'modi' },
          { uuid: 2, name: 'ipsum' },
          { uuid: 3, name: 'architecto' }
        ]
      }
    },
    methods: {
      filterMember (params) {
        action('Run')(params)
      }
    },
    template: `
      <v-container>
        <add-store-filter
          :types="types"
          @runFilter="filterMember"
        />
      </v-container>
    `
  }))
