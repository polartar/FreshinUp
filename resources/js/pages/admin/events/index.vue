<template>
  <div>
    <v-flex
      d-flex
      align-center
      justify-space-between
      ma-3
    >
      <v-layout
        flex
        align-center
      >
        <h2 class="white--text mr-3">
          {{ pageTitle }}
        </h2>
        <v-btn
          slot="activator"
          color="white"
          @click="eventNew"
        >
          <span class="primary--text">Add New Event</span>
        </v-btn>
      </v-layout>
      <v-flex
        text-xs-right
        sm2
        xs12
      >
        <v-select
          v-model="view"
          :items="views"
          single-line
          solo
          flat
          hide-details
        />
      </v-flex>
    </v-flex>
    <v-divider />
    <template
      v-if="view === 1"
    >
      <filter-sorter
        v-if="!isLoading"
        :statuses="statuses"
        :types="eventTypes"
        @runFilter="filterEvents"
      />
      <event-list
        v-if="!isLoading"
        :events="events"
        :statuses="statuses"
        :is-loading="isLoading || isLoadingList"
        :rows-per-page="pagination.rowsPerPage"
        :page="pagination.page"
        :total-items="pagination.totalItems"
        :sort-by="sorting.sortBy"
        :descending="sorting.descending"
        @paginate="onPaginate"
        @manage-edit="eventEdit"
        @manage-delete="deleteSingle"
        @manage-multiple-delete="multipleDelete"
        @change-status="changeStatusSingle"
        @change-status-multiple="changeStatusMultiple"
      />
    </template>
    <v-layout
      v-else-if="view === 2"
      row
      flex
    >
      <v-flex
        sm3
        xs12
        ml-3
        mt-3
      >
        <v-card>
          <filter-sorter-for-calendar
            v-if="!isLoading"
            :statuses="statuses"
            @runFilter="filterEvents"
          />
        </v-card>
      </v-flex>
      <v-flex
        sm9
        xs12
        ma-3
      >
        <v-card>
          <event-calendar
            v-if="!isLoading"
            :events="events"
            :date="calendarDefaultDate"
            :year-range="calendarYearRange"
            @click-event="eventEdit"
          />
        </v-card>
      </v-flex>
    </v-layout>
    <v-dialog
      v-model="deleteDialog"
      max-width="500"
    >
      <simple-confirm
        :class="{ 'deleting': deletablesProcessing }"
        :title="deleteDialogTitle"
        ok-label="Yes"
        cancel-label="No"
        @ok="onSubmitDelete"
        @cancel="onCancelDelete"
      >
        <div class="py-5 px-2">
          <template v-if="deletablesProcessing">
            <div class="text-xs-center">
              <p class="subheading">
                Processing, please wait...
              </p>
              <v-progress-circular
                :rotate="-90"
                :size="200"
                :width="15"
                :value="deletablesProgress"
                color="primary"
              >
                {{ deletablesStatus }}
              </v-progress-circular>
            </div>
          </template>
          <template v-else>
            <p class="subheading">
              <span v-if="deletables.length < 2">Event</span>
              <span v-else> Events</span>
              : {{ deleteTemp | formatDeleteTitles }}
            </p>
          </template>
        </div>
      </simple-confirm>
    </v-dialog>
  </div>
</template>
<script>
import get from 'lodash/get'
import moment from 'moment'
import { mapActions, mapGetters, mapState } from 'vuex'
import { deletables } from 'fresh-bus/components/mixins/Deletables'
import FilterSorter from '~/components/events/FilterSorter.vue'
import FilterSorterForCalendar from '~/components/events/FilterSorterForCalendar.vue'
import EventList from '~/components/events/EventList.vue'
import EventCalendar from '~/components/events/EventCalendar.vue'
import SimpleConfirm from 'fresh-bus/components/SimpleConfirm.vue'

const INCLUDE = [
  'status',
  'host',
  'location',
  'manager',
  'type',
  'venue'
]

export default {
  layout: 'admin',
  components: {
    EventList,
    EventCalendar,
    FilterSorter,
    FilterSorterForCalendar,
    SimpleConfirm
  },
  filters: {
    formatDeleteTitles (value) {
      return value.map(item => item.name).join(', ')
    }
  },
  mixins: [deletables],
  data () {
    const currentMoment = moment()
    return {
      pageTitle: 'Events',
      deleteDialog: false,
      deleteTemp: [],
      deletablesProcessing: false,
      deletablesProgress: 0,
      deletablesStatus: '',
      lastFilterParams: {},
      calendarDefaultDate: currentMoment.format('YYYY-MM-DD'),
      calendarYearRange: [currentMoment.year() - 2, currentMoment.year() + 2],
      view: 1,
      views: [
        { value: 1, text: 'List view' },
        { value: 2, text: 'Calendar view' }
      ]
    }
  },
  computed: {
    isLoadingList () {
      return get(this.$store, 'state.events.pending.items', true)
    },
    ...mapGetters(['currentUser']),
    ...mapGetters('events', {
      events: 'items',
      pagination: 'pagination',
      sorting: 'sorting',
      sortBy: 'sortBy'
    }),
    ...mapGetters('eventStatuses', { 'statuses': 'items' }),
    ...mapGetters('eventTypes', { 'eventTypes': 'items' }),
    ...mapGetters('page', ['isLoading']),
    ...mapState('events', ['sortables']),
    deleteDialogTitle () {
      return this.deleteTemp.length < 2 ? 'Are you sure you want to delete this event?' : 'Are you sure you want to delete the following events?'
    },
    role () {
      if (this.currentUser.isAdmin) return 'admin'
      if (this.currentUser.type === 1) return 'supplier'
      if (this.currentUser.type === 2) return 'host'
      return 'admin'
    }
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    }),
    eventNew () {
      this.$router.push({ path: '/admin/events/new' })
    },
    eventEdit (event) {
      this.$router.push({ path: '/admin/events/' + event.uuid + '/edit' })
    },
    deleteSingle (event) {
      this.deleteTemp = [event]
      this.deleteDialog = true
    },
    multipleDelete (events) {
      this.deleteTemp = events
      this.deleteDialog = true
    },
    changeStatusSingle (statusId, event) {
      this.$store.dispatch('events/patchItem', {
        data: { status_id: statusId },
        params: { id: event.uuid }
      }).then(() => {
        this.filterEvents(this.lastFilterParams)
      })
    },
    changeStatusMultiple (statusId, events) {
      events.forEach((event) => {
        this.changeStatusSingle(statusId, event)
      })
    },
    async onSubmitDelete () {
      this.deletablesProcessing = true
      this.deletablesProgress = 0
      this.deletablesStatus = ''
      let dispatcheables = []

      this.deleteTemp.forEach((event) => {
        dispatcheables.push(this.$store.dispatch('events/deleteItem', {
          getItems: false,
          params: { id: event.uuid }
        }))
      })

      let chunks = this.chunk(dispatcheables, this.deleteTempParrallelRequest)
      let doneCount = 0

      for (let i in chunks) {
        await Promise.all(chunks[i])
        doneCount += chunks[i].length
        this.deleteTempStatus = doneCount + ' / ' + this.deleteTemp.length + ' Done'
        this.deleteTempProgress = doneCount / this.deleteTemp.length * 100
        await this.sleep(this.deletablesSleepTime)
      }

      this.filterEvents(this.lastFilterParams)
      await this.sleep(500)
      this.deletablesProcessing = false
      this.deleteDialog = false
    },
    onCancelDelete () {
      this.deleteDialog = false
      this.deleteTemp = []
    },
    onPaginate (value) {
      this.$store.dispatch('events/setPagination', value)
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE } })
    },
    filterEvents (params) {
      this.lastFilterParams = params
      this.$store.dispatch('events/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams
      })
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE } })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('events/setFilters', {
      ...vm.$route.query,
      ...this.lastFilterParams
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('eventStatuses/getItems'),
      vm.$store.dispatch('eventTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => {
        vm.setPageLoading(false)
      })
  }
}
</script>
