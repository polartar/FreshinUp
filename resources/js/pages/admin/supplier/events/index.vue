<script>
import Page, { INCLUDE } from '~/pages/admin/events/index.vue'
import { mapGetters } from 'vuex'
export default {
  extends: Page,
  computed: {
    ...mapGetters('suppliers/events', {
      events: 'items',
      pagination: 'pagination',
      sorting: 'sorting',
      sortBy: 'sortBy',
      sortables: 'sortables'
    })
  },
  methods: {
    eventNew () {
      this.$router.push({ path: '/admin/supplier/events/new' })
    },
    eventEdit (event) {
      this.$router.push({ path: '/admin/supplier/events/' + event.uuid + '/edit' })
    },
    onPaginate (value) {
      this.$store.dispatch('suppliers/events/setPagination', value)
      this.$store.dispatch('suppliers/events/getItems', {
        params: {
          include: INCLUDE,
          supplierId: this.currentUser.uuid
        }
      })
    },
    filterEvents (params) {
      this.lastFilterParams = params
      this.$store.dispatch('suppliers/events/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams
      })
      this.$store.dispatch('suppliers/events/getItems', {
        params: {
          include: INCLUDE,
          supplierId: this.currentUser.uuid
        }
      })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('suppliers/events/setFilters', {
      ...vm.$route.query,
      ...this.lastFilterParams
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('eventStatuses/getItems'),
      vm.$store.dispatch('eventTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => {
        vm.setPageLoading(false)
      })
  }
}
</script>
