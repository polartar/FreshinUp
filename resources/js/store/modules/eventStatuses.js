import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default ({ items, item }) => {
  return makeRestStore('foodfleet/event-statuses', { items, item })
}
