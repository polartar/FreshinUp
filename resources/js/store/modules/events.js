import makeRestStore from 'fresh-bus/store/utils/makeRestStore'
import omit from 'lodash/omit'

export default ({ items, item }) => {
  const store = makeRestStore(
    'events',
    { items, item },
    {
      itemsPath: () => `/foodfleet/events`,
      itemPath: ({ id }) => `/foodfleet/events/${id}`
    }
  )
  const __createItem = store.actions.createItem
  store.actions = {
    ...store.actions,
    createItem (context, payload) {
      payload.data = omit(payload.data, ['event_recurring_checked', 'event_tags', 'host', 'venue', 'manager', 'status'])
      return __createItem(context, payload)
    }
  }

  return {
    namespaced: true,
    ...store,
    modules: {
      stores: makeRestStore(
        'stores',
        { items, item },
        {
          itemsPath: ({ eventId }) => `/foodfleet/events/${eventId}/stores`,
          itemPath: ({ eventId, id }) => `/foodfleet/events/${eventId}/stores/${id}`
        }
      )
    }
  }
}
