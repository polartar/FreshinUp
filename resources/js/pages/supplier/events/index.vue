
<script>
import { mapActions, mapGetters } from 'vuex'
import events from '~/pages/admin/events/index.vue'

const INCLUDE = [
  'status',
  'host',
  'location',
  'manager',
  'type',
  'venue'
]

export default {
  extends: events,
  computed: {
    ...mapGetters(['currentUser'])
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    }),
    onPaginate (value) {
      this.$store.dispatch('events/setPagination', value)
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE, manager_uuid: this.currentUser.uuid } })
    },
    filterEvents (params) {
      this.lastFilterParams = params
      this.$store.dispatch('events/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams,
        manager_uuid: this.currentUser.uuid
      })
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE, manager_uuid: this.currentUser.uuid } })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('events/setFilters', {
      ...vm.$route.query,
      ...this.lastFilterParams,
      manager_uuid: vm.currentUser.uuid
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('eventStatuses/getItems'),
      vm.$store.dispatch('eventTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => {
        vm.setPageLoading(false)
      })
  }
}
</script>
