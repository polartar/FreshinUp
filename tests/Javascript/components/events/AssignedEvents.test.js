import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/events/AssignedEvents.vue'
import { FIXTURE_EVENTS } from 'tests/__data__/events'
import { FIXTURE_EVENT_STATUSES } from 'tests/__data__/eventStatuses'

describe('Assigned Events component', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('events set', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          statuses: FIXTURE_EVENT_STATUSES,
          events: FIXTURE_EVENTS
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('events empty', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          statuses: FIXTURE_EVENT_STATUSES,
          events: []
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })

    test('emit actions', () => {
      const wrapper = shallowMount(Component, {
        localVue
      })

      const mockEvent = { id: 1, title: 'mock title', status: 1 }
      wrapper.vm.viewEvent(mockEvent)
      expect(wrapper.emitted()['viewEvent']).toBeTruthy()

      wrapper.vm.sort({ id: 'name', label: 'Name' })
      expect(wrapper.emitted()['sort']).toBeTruthy()

      wrapper.vm.searchInput('test')
      expect(wrapper.emitted()['searchInput']).toBeTruthy()
    })
  })
})
