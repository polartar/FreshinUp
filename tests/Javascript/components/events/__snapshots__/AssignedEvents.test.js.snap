// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assigned Events component Snapshots events empty 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__title px-3"
  >
    <h3
      class="primary--text"
    >
      
      Assigned Events
    
    </h3>
  </div>
   
  <hr />
   
  <div
    class="v-card__text text-xs-center pa-5 primary--text"
  >
    
    Event data will populate once your restaurant is assigned.
  
  </div>
</div>
`;

exports[`Assigned Events component Snapshots events set 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__title px-3"
  >
    <h3
      class="primary--text"
    >
      
      Assigned Events
    
    </h3>
  </div>
   
  <hr />
   
  <div
    class="v-card__text ma-2"
  >
    <div
      class="filter-transparent v-card v-sheet theme--light transparent"
      icon="search"
    >
      <form
        class="v-form"
        novalidate="novalidate"
      >
        <div
          class="container filter-sorter-container fluid"
        >
           
          <div
            class="layout filter-sorter-main-layout row wrap"
          >
            <!---->
             
            <div
              class="flex grow"
            >
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Filter by event name"
                        type="text"
                      />
                    </div>
                    <div
                      class="v-input__append-inner"
                    >
                      <div
                        class="v-input__icon v-input__icon--append"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          search
                        </i>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <!---->
             
            <!---->
             
          </div>
           
          <hr
            class="v-divider theme--light"
          />
           
          <div
            class="layout filter-sorter-expanded-layout align-center justify-center row fill-height"
          >
            <div
              class="flex"
            />
             
            <div
              class="flex text-no-wrap"
            >
              <button
                class="v-btn v-btn--large theme--light"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
            Clear Filters
          
                </div>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
     
    <div
      class="event-list-table mt-4"
    >
      <div
        class="v-table__overflow"
      >
        <table
          class="v-datatable v-table theme--light"
        >
          <thead>
            <tr>
              <th
                aria-label="Status: Not sorted. Activate to sort ascending."
                aria-sort="none"
                class="column sortable text-xs-left"
                role="columnheader"
                scope="col"
                tabindex="0"
              >
                <span>
                  Status
                </span>
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                  style="font-size: 16px;"
                >
                  arrow_upward
                </i>
              </th>
              <th
                aria-label="Name / Category: Not sorted. Activate to sort ascending."
                aria-sort="none"
                class="column sortable text-xs-left"
                role="columnheader"
                scope="col"
                tabindex="0"
              >
                <span>
                  Name / Category
                </span>
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                  style="font-size: 16px;"
                >
                  arrow_upward
                </i>
              </th>
              <th
                aria-label="Date / Venue: Not sorted. Activate to sort ascending."
                aria-sort="none"
                class="column sortable text-xs-left"
                role="columnheader"
                scope="col"
                tabindex="0"
              >
                <span>
                  Date / Venue
                </span>
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                  style="font-size: 16px;"
                >
                  arrow_upward
                </i>
              </th>
              <th
                aria-label="Managed By: Not sorted."
                aria-sort="none"
                class="column text-xs-left"
                role="columnheader"
                scope="col"
              >
                <span>
                  Managed By
                </span>
              </th>
              <th
                aria-label="Customer: Not sorted."
                aria-sort="none"
                class="column text-xs-left"
                role="columnheader"
                scope="col"
              >
                <span>
                  Customer
                </span>
              </th>
              <th
                aria-label="Manage: Not sorted."
                aria-sort="none"
                class="column text-xs-left"
                role="columnheader"
                scope="col"
              >
                <span>
                  Manage
                </span>
              </th>
            </tr>
            <tr
              class="v-datatable__progress"
            >
              <th
                class="column"
                colspan="6"
              />
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="select-td"
              >
                <span
                  class="v-btn v-btn--disabled accent accenttext--text"
                >
                  Draft
                </span>
              </td>
               
              <td>
                <div
                  class="subheading primary--text"
                >
                  
            accusantium
          
                </div>
                 
                <!---->
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Oct 10 - 12 / 2019 11:04 AM
          
                </div>
                 
                <div
                  class="grey--text"
                >
                  
            @New Hattie
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Demo Admin
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Swift-Wehner
          
                </div>
              </td>
               
              <td>
                <button
                  class="v-btn theme--light success success-text--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            View
          
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td
                class="select-td"
              >
                <span
                  class="v-btn v-btn--disabled warning warningtext--text"
                >
                  Pending
                </span>
              </td>
               
              <td>
                <div
                  class="subheading primary--text"
                >
                  
            saepe
          
                </div>
                 
                <!---->
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Oct 10 - 12 / 2019 11:04 AM
          
                </div>
                 
                <div
                  class="grey--text"
                >
                  
            @Baronhaven
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Level 2 User
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Goodwin-Carroll
          
                </div>
              </td>
               
              <td>
                <button
                  class="v-btn theme--light success success-text--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            View
          
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td
                class="select-td"
              >
                <span
                  class="v-btn v-btn--disabled success successtext--text"
                >
                  Confirmed
                </span>
              </td>
               
              <td>
                <div
                  class="subheading primary--text"
                >
                  
            ipsum
          
                </div>
                 
                <!---->
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Oct 10 - 12 / 2019 11:04 AM
          
                </div>
                 
                <div
                  class="grey--text"
                >
                  
            @Lake Lavernehaven
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Level 1 User
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Carter-Green
          
                </div>
              </td>
               
              <td>
                <button
                  class="v-btn theme--light success success-text--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            View
          
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td
                class="select-td"
              >
                <span
                  class="v-btn v-btn--disabled secondary secondarytext--text"
                >
                  Past
                </span>
              </td>
               
              <td>
                <div
                  class="subheading primary--text"
                >
                  
            quibusdam
          
                </div>
                 
                <!---->
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Oct 10 - 12 / 2019 11:04 AM
          
                </div>
                 
                <div
                  class="grey--text"
                >
                  
            @Lake Lavernehaven
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Level 1 User
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Carter-Green
          
                </div>
              </td>
               
              <td>
                <button
                  class="v-btn theme--light success success-text--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            View
          
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td
                class="select-td"
              >
                <span
                  class="v-btn v-btn--disabled accent accenttext--text"
                >
                  Cancelled
                </span>
              </td>
               
              <td>
                <div
                  class="subheading primary--text"
                >
                  
            lorem
          
                </div>
                 
                <!---->
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Oct 10 - 12 / 2019 11:04 AM
          
                </div>
                 
                <div
                  class="grey--text"
                >
                  
            @Lake Lavernehaven
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Level 1 User
          
                </div>
              </td>
               
              <td>
                <div
                  class="grey--text"
                >
                  
            Carter-Green
          
                </div>
              </td>
               
              <td>
                <button
                  class="v-btn theme--light success success-text--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            View
          
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="v-datatable v-table theme--light"
      >
        <div
          class="v-datatable__actions"
        >
          <div
            class="v-datatable__actions__select"
          >
            Rows per page:
            <div
              class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
              role="combobox"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-select__slot"
                  >
                    <div
                      class="v-select__selections"
                    >
                      <div
                        class="v-select__selection v-select__selection--comma"
                      >
                        10
                      </div>
                      <input
                        aria-label="Rows per page:"
                        aria-readonly="false"
                        autocomplete="on"
                        readonly="readonly"
                        type="text"
                      />
                    </div>
                    <div
                      class="v-input__append-inner"
                    >
                      <div
                        class="v-input__icon v-input__icon--append"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          arrow_drop_down
                        </i>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-menu"
                  >
                    <div
                      class="v-menu__content theme--light v-menu__content--auto "
                      style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                    >
                      <div
                        class="v-select-list v-card theme--light"
                      >
                        <div
                          class="v-list theme--light"
                          role="list"
                        >
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  5
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            class="primary--text"
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  10
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  15
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  25
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  30
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  50
                                </div>
                              </div>
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-datatable__actions__range-controls"
          >
            <div
              class="v-datatable__actions__pagination"
            >
              1-10 of 10
            </div>
            <button
              aria-label="Previous page"
              class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
              disabled="disabled"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  chevron_left
                </i>
              </div>
            </button>
            <button
              aria-label="Next page"
              class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
              disabled="disabled"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                <i
                  aria-hidden="true"
                  class="v-icon material-icons theme--light"
                >
                  chevron_right
                </i>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
